I have below api, I need to do code review please provide me the improvements comments 

@app.route('/users', methods=['POST'])
def add_user() -> ResponseReturnValue:
    """
    Summary: Add user to the Cognito user pool.

    Returns:
        (ResponseReturnValue): JSON response with message.
    """
    data = request.json

    if not data:
        return jsonify({'message': 'No data provided'}), 400
    try:
        USER_POOL_ID = get_user_pool_id()  # noqa: N806
        role_info = get_role_code(data['role'])
        if 'error' in role_info:
            return jsonify({'message': role_info['error']}), 400
        role_code = role_info['role_code']
        cognito_client.admin_create_user(
            UserPoolId=USER_POOL_ID,
            Username=data['email'],
            UserAttributes=[
                {"Name": "email", "Value": data['email']},
                {"Name": "email_verified", "Value": "true"},
                {"Name": "custom:emp_id", "Value": data['emp_id']},
                {
                    "Name": "custom:role_code",
                    "Value": role_code,
                },
                {"Name": "custom:name", "Value": data['name']},
                {"Name": "custom:projects", "Value": str(data['projects'])},
                {"Name": "custom:upn_str", "Value": data['upn']},
            ],
            DesiredDeliveryMediums=['EMAIL'],
        )
        return jsonify({'message': 'User created successfully'}), 201
    except ClientError as e:
        error_message = e.response['Error']['Message']
        if e.response['Error']['Code'] == 'UsernameExistsException':
            return jsonify({'message': 'User already exists'}), 409
        return (
            jsonify({'message': f"Error creating user: {error_message}"}),
            500,
        )
